### ISSUE TITLE
自己的数据，定制训练UIE，模型提取将所有句子的句首部分信息完全丢失

### ISSUE THREAD

**Issue by ymzx** [PaddlePaddle/PaddleNLP#2251]
环境： linux/python3.8+/PaddleNLP-develop/paddlenlp 2.3.0/paddle2.2.1
例如 "脑内多发小缺血灶。双侧筛窦慢性炎症。双侧颈深上间隙淋巴结反应性增生。"
“脑内多发小缺血灶” 所有信息遗漏，其它部分的信息提取正确。
而且信息提取丢失呈现一定的规律，就是所有句子句首信息都提取不到，句子中部和后部的信息完成能提取到，很好奇这个是可能是因为什么原因？
完全按照https://github.com/PaddlePaddle/PaddleNLP/tree/develop/model_zoo/uie做的UIE finetune，
谢谢！
采用doccano数据标注
![image](https://user-images.githubusercontent.com/29808550/169265965-c25fa290-f944-42a7-ab49-e52329c3ef9a.png)
{"id": 30, "text": "脑内多发小缺血灶；轻度脑白质变性。", "relations": [], "entities": [{"id": 166, "start_offset": 0, "end_offset": 2, "label": "part"}, {"id": 167, "start_offset": 4, "end_offset": 8, "label": "disease"}, {"id": 168, "start_offset": 9, "end_offset": 16, "label": "disease"}]}

其它样本举例
胃窦部慢性炎症，请结合胃镜随访。
{"id": 37, "text": "胃窦部慢性炎症，请结合胃镜随访。", "relations": [], "entities": [{"id": 200, "start_offset": 0, "end_offset": 3, "label": "part"}, {"id": 201, "start_offset": 3, "end_offset": 7, "label": "disease"}, {"id": 202, "start_offset": 8, "end_offset": 15, "label": "suggest"}]}
前列腺钙化灶。右肾小结石。
{"id": 27, "text": "前列腺钙化灶。右肾小结石。", "relations": [], "entities": [{"id": 153, "start_offset": 0, "end_offset": 3, "label": "part"}, {"id": 154, "start_offset": 3, "end_offset": 6, "label": "disease"}, {"id": 155, "start_offset": 7, "end_offset": 9, "label": "part"}, {"id": 156, "start_offset": 9, "end_offset": 12, "label": "disease"}]}

训练脚本
python finetune.py \
    --train_path "./data/train.txt" \
    --dev_path "./data/dev.txt" \
    --save_dir "./checkpoint" \
    --learning_rate 1e-5 \
    --batch_size 16 \
    --max_seq_len 512 \
    --num_epochs 100 \
    --model "uie-base" \
    --seed 1000 \
    --logging_steps 10 \
    --valid_steps 100 \
    --device "gpu"

预测
from pprint import pprint
from paddlenlp import Taskflow
schema = ['part', 'disease', 'evidence', 'result', 'suggest'] # Define the schema for entity extraction
ie = Taskflow('information_extraction', schema=schema, task_path=r'E:\PaddleNLP-develop\model_zoo\uie\checkpoint\model_1400') #  ['uie-base', 'uie-medical-base', 'uie-tiny']
pprint(ie("左侧咽旁间隙异常信号灶，FDG未见异常摄取，建议MR增强随访复查(3个月)。右侧颌下淋巴结FDG摄取增高，考虑为淋巴结反应性增生。"))

结果
[{'disease': [{'end': 64,
               'probability': 0.9999687673167728,
               'start': 56,
               'text': '淋巴结反应性增生'}],
  'evidence': [{'end': 21,
                'probability': 0.9999477869888551,
                'start': 12,
                'text': 'FDG未见异常摄取'},
               {'end': 11,
                'probability': 0.9999557738387068,
                'start': 6,
                'text': '异常信号灶'},
               {'end': 52,
                'probability': 0.9999815226331066,
                'start': 45,
                'text': 'FDG摄取增高'}],
  'part': [{'end': 45,
            'probability': 0.9999802113476477,
            'start': 38,
            'text': '右侧颌下淋巴结'}],
  'result': [{'end': 64,
              'probability': 0.9999666215932024,
              'start': 53,
              'text': '考虑为淋巴结反应性增生'}],
  'suggest': [{'end': 37,
               'probability': 0.9999763966967521,
               'start': 22,
               'text': '建议MR增强随访复查(3个月)'}]}]

“左侧咽旁间隙”这个就漏掉了。

如果输入为“胃窦部慢性炎症，请结合胃镜随访。”
输出：
[{'suggest': [{'end': 15,
               'probability': 0.9999722243814233,
               'start': 8,
               'text': '请结合胃镜随访'}]}]
胃窦部慢性炎症漏掉了

**Comment by wawltor** [comment]
能看一下您怎么标注数据的吗？

**Comment by ymzx** [comment]
> 

已在问题中更新标注信息

**Comment by kahitomi** [comment]
信息太少了~ 可能先排查一下训练数据是不是给对了
第一步：看看训练好的模型在训练数据上是不是也有这个表现？
有->标注平台问题 or 数据转换脚本问题，看转换后的训练数据是不是全的
没有->确认是不是100%句子句首信息都抽取不到，然后是不是给一下最少量可以复现的训练数据和训练参数

**Comment by ymzx** [comment]
> 

在训练数据上也有一样头部信息抽取丢失的表现，看转换后的训练数据是全的，最少量可以复现的训练数据和训练参数已给。

**Comment by kahitomi** [comment]
相关同学正在看了，有一点要先说一下，schema里面的标签名（prompt）在现在的版本最好用中文~ 咱们现在开源的是中文模型。比如part->身体部位

**Comment by ymzx** [comment]
嗯好的，感谢，因为目前还没梳理paddlenlp的源码逻辑，所以相关的优缺点以及需要注意的地方并不清楚，我改成中文再试试。


------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "PaddlePaddle/PaddleNLP"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年5月19日(星期四) 晚上9:51
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [PaddlePaddle/PaddleNLP] 自己的数据，定制训练UIE，模型提取将所有句子的句首部分信息完全丢失 (Issue #2251)





 
相关同学正在看了，有一点要先说一下，schema里面的标签名（prompt）在现在的版本最好用中文~ 咱们现在开源的是中文模型。比如part-&gt;身体部位

**Comment by wawltor** [comment]
> 嗯好的，感谢，因为目前还没梳理paddlenlp的源码逻辑，所以相关的优缺点以及需要注意的地方并不清楚，我改成中文再试试。
> […](#)
> ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: "PaddlePaddle/PaddleNLP" ***@***.***&gt;; 发送时间:&nbsp;2022年5月19日(星期四) 晚上9:51 ***@***.***&gt;; ***@***.******@***.***&gt;; 主题:&nbsp;Re: [PaddlePaddle/PaddleNLP] 自己的数据，定制训练UIE，模型提取将所有句子的句首部分信息完全丢失 (Issue #2251) 相关同学正在看了，有一点要先说一下，schema里面的标签名（prompt）在现在的版本最好用中文~ 咱们现在开源的是中文模型。比如part-&gt;身体部位 — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you authored the thread.Message ID: ***@***.***&gt;

改成中文效果咋样的了？

**Comment by ymzx** [comment]
> > 嗯好的，感谢，因为目前还没梳理paddlenlp的源码逻辑，所以相关的优缺点以及需要注意的地方并不清楚，我改成中文再试试。
> > […](#)
> > ------------------ 原始邮件 ------------------ 发件人: "PaddlePaddle/PaddleNLP" _**@**_._**>; 发送时间: 2022年5月19日(星期四) 晚上9:51 _**@**_.**_>; _**@**_._**_**@**_.**_>; 主题: Re: [PaddlePaddle/PaddleNLP] 自己的数据，定制训练UIE，模型提取将所有句子的句首部分信息完全丢失 (Issue #2251) 相关同学正在看了，有一点要先说一下，schema里面的标签名（prompt）在现在的版本最好用中文~ 咱们现在开源的是中文模型。比如part->身体部位 — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you authored the thread.Message ID: _**@**_.***>
> 
> 改成中文效果咋样的了？

改成中文就OK了，能简单解释下么？😎

**Comment by wawltor** [comment]
> > > 嗯好的，感谢，因为目前还没梳理paddlenlp的源码逻辑，所以相关的优缺点以及需要注意的地方并不清楚，我改成中文再试试。
> > > […](#)
> > > ------------------ 原始邮件 ------------------ 发件人: "PaddlePaddle/PaddleNLP" _**@**_._**>; 发送时间: 2022年5月19日(星期四) 晚上9:51 _**@**_.**_>; _**@**_._**_**@**_.**_>; 主题: Re: [PaddlePaddle/PaddleNLP] 自己的数据，定制训练UIE，模型提取将所有句子的句首部分信息完全丢失 (Issue #2251) 相关同学正在看了，有一点要先说一下，schema里面的标签名（prompt）在现在的版本最好用中文~ 咱们现在开源的是中文模型。比如part->身体部位 — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you authored the thread.Message ID: _**@**_.***>
> > 
> > 
> > 改成中文效果咋样的了？
> 
> 改成中文就OK了，能简单解释下么？😎

因为UIE是基于中文预训练模型和prompt来做的，prompt偏向于中文

**Comment by ymzx** [comment]
好的 学习了。


------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "PaddlePaddle/PaddleNLP"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年5月20日(星期五) 下午5:18
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [PaddlePaddle/PaddleNLP] 自己的数据，定制训练UIE，模型提取将所有句子的句首部分信息完全丢失 (Issue #2251)





    
嗯好的，感谢，因为目前还没梳理paddlenlp的源码逻辑，所以相关的优缺点以及需要注意的地方并不清楚，我改成中文再试试。
 …
 ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: "PaddlePaddle/PaddleNLP" @.&gt;; 发送时间:&nbsp;2022年5月19日(星期四) 晚上9:51 _@**._&gt;; ***@***.***@._&gt;; 主题:&nbsp;Re: [PaddlePaddle/PaddleNLP] 自己的数据，定制训练UIE，模型提取将所有句子的句首部分信息完全丢失 (Issue #2251) 相关同学正在看了，有一点要先说一下，schema里面的标签名（prompt）在现在的版本最好用中文~ 咱们现在开源的是中文模型。比如part-&gt;身体部位 — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you authored the thread.Message ID: _@_.*&gt;
  
改成中文效果咋样的了？
  
改成中文就OK了，能简单解释下么？😎
  
因为UIE是基于中文预训练模型和prompt来做的，prompt偏向于中文
